frontend/public/components/storage/attach-storage.tsx

```diff
@@ -40,306 +27,309 @@ const PVCDropdown: React.SFC<PVCDropdownProps> = props => {
-    const template = resourceObj.spec.template;
-
-    // For each container in the pod spec, add the new volume mount.
-    _.each(template.spec.containers, container => {
-      if (!this.isContainerSelected(container)) {
-        return;
-      }
-      if (!container.volumeMounts) {
-        container.volumeMounts = [];
-      }
-      container.volumeMounts.push({
-        name: volumeName,
-        mountPath,
-        subPath,
-        readOnly: mountAsReadOnly,
-      });
-    });
+      readOnly: mountAsReadOnly,
+    };
 
-    this.setState({ inProgress: true });
-    this.createPVCIfNecessary().then(claimName => {
-      // add new volume to the pod template if not preent
-      if (!volumeAlreadyMounted) {
-        template.spec.volumes = template.spec.volumes || [];
-        template.spec.volumes.push({
-          name: volumeName,
-          persistentVolumeClaim: {
-            claimName,
-          },
-        });
+    const containers: ContainerSpec[] = _.get(obj, 'spec.template.spec.containers', []);
+    const patches = containers.reduce((patch, container, i) => {
+      // Only add to selected containers
+      if (isContainerSelected(container)) {
+        if (_.isEmpty(container.volumeMounts)) {
+          patch.push({op: 'add', path: `/spec/template/spec/containers/${i}/volumeMounts`, value: [mount]});
+        } else {
+          patch.push({op: 'add', path: `/spec/template/spec/containers/${i}/volumeMounts/-`, value: mount});
+        }
       }
-      k8sUpdate(kindObj, resourceObj, metadata.namespace, metadata.name).then(
-        resource => {
-          this.setState({ inProgress: false });
-          history.push(resourceObjPath(resource, referenceFor(resource)));
-        },
-        err => this.setState({ error: err.message, inProgress: false })
-      );
-    }, err => this.setState({ error: err.message, inProgress: false }));
+      return patch;
+    }, []);
+    const volume = {
+      name: volumeName,
+      persistentVolumeClaim: {
+        claimName: pvClaimName,
+      },
+    };
 
+    if (!volumeAlreadyMounted) {
+      const existingVolumes = _.get(obj, 'spec.template.spec.volumes');
+      const volumePatch = _.isEmpty(existingVolumes)
+        ? {op: 'add', path: '/spec/template/spec/volumes', value: [volume]}
+        : {op: 'add', path: '/spec/template/spec/volumes/-', value: volume};
+      return [ ...patches, volumePatch ];
+    }
+    return patches;
   };
 
-  onPVCChange = newPVCObj => {
-    this.setState({ newPVCObj });
-    this.updateVolumeName(_.get(newPVCObj, 'metadata.name', ''));
+  const save = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    if (useContainerSelector && selectedContainers.length === 0) {
+      setError('You must choose at least one container to mount to.');
+      return;
+    }
+    setInProgress(true);
+    createPVCIfNecessary().then((pvClaimName: string) => {
+      k8sPatch(kindObj, obj, getVolumePatches(pvClaimName)).then(
```

Comment:
Very important we return the promise here or we won't show errors

```suggestion
      return k8sPatch(kindObj, obj, getVolumePatches(pvClaimName)).then(
```

Reply:
Ok, I had to drill into the k8sPatch code to see that it returns a promise. Good catch!.  It appears that this was a bug with k8sUpdate as well then?

Reply:
I think it was OK until we removed the duplicate error handling, but having one error handler is fine as long as we return the promise here.

Reply:
Code is updated